playbooks/reboot_cluster.yml

```diff
@@ -1,48 +1,123 @@
 ---
-- hosts: all:!db:!vyatta-*
+- hosts: controller
   serial: 1
   tasks:
   - name: Reboot
     command: shutdown -r now
     async: 0
     poll: 0
-    ignore_errors: true
+    failed_when: false
 
   - name: Wait for SSH to come back
     wait_for: host={{ ansible_default_ipv4.address }}
               port={{ ansible_ssh_port|default(22) }}
               delay=15
     delegate_to: "{{ groups['controller']|first }}"
 
+  - name: cluster health check
+    command: /etc/sensu/plugins/percona-cluster-size.rb -d /root/.my.cnf --expected 3 --criticality critical
+    register: cstat
+    until: cstat | succeeded
+    retries: 5
+    run_once: true
+    delegate_to: "{{ item }}"
+    with_items: "{{ play_hosts }}"
```

Comment:
I don't think you need to do this on all the hosts, since the cluster status depends on all the hosts. Checking on a single host give you the info you need to proceed.